src/sentry/static/sentry/app/views/performance/traceDetails/content.tsx

```diff
@@ -55,6 +70,53 @@ class TraceDetailsContent extends React.Component<Props> {
     return <LoadingError message={t('The trace you are looking for was not found.')} />;
   }
 
+  handleTransactionFilter = (searchQuery: string) => {
+    this.setState({searchQuery: searchQuery || undefined}, this.filterTransactions);
+  };
+
+  filterTransactions = async () => {
+    const {trace} = this.props;
+    const {filteredTransactionIds, searchQuery} = this.state;
+
+    if (!searchQuery || trace === null) {
+      if (filteredTransactionIds !== undefined) {
+        this.setState({
+          filteredTransactionIds: undefined,
+        });
+      }
+      return;
+    }
+
+    // TODO(tonyx): this should perform a fuzzy search
+    const matched = filterTrace(trace, ({transaction}) =>
+      transaction.includes(searchQuery)
```

Comment:
nit: I know we're switching to fuzzy, but could this be case insensitive?

Reply:
nit2: (maybe not a nit though, i just know we're adding fuzzy search so iffy on whether we need to do in this PR) But we need to search the transaction.op too to match how span's do this

Reply:
I can update this to do a case insensitive search on the transaction op and name until fuzzy search is implemented.

Reply:
Updated in 43641fa